google_compute_engine/metadata_scripts/script_retriever.py

```diff
@@ -50,29 +55,40 @@ def _DownloadGsUrl(self, url, dest_dir):
     Returns:
       string, the path to the file storing the metadata script.
     """
-    try:
-      subprocess.check_call(
-          ['which', 'gsutil'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    except subprocess.CalledProcessError:
-      self.logger.warning(
-          'gsutil is not installed, cannot download items from Google Storage.')
-      return None
-
     dest_file = tempfile.NamedTemporaryFile(dir=dest_dir, delete=False)
     dest_file.close()
     dest = dest_file.name
 
-    self.logger.info('Downloading url from %s to %s using gsutil.', url, dest)
+    self.logger.info(
+        'Downloading url from %s to %s using authentication token.',
+        url, dest)
     try:
-      subprocess.check_call(['gsutil', 'cp', url, dest])
+      if not self._token:
+        tok_url = '%(metadata)s/instance/service-accounts/default/token' % {
+            'metadata': 'http://metadata.google.internal/computeMetadata/v1',
+        }
+        request = urlrequest.Request(tok_url)
+        request.add_unredirected_header('Metadata-Flavor', 'Google')
+        # converts the stringified dictionary of the response to a dictionary
```

Comment:
All comments are sentences - pease capitalize the first letter and end in a period.